apps/src/templates/instructions/utils.js

```diff
@@ -114,8 +114,15 @@ function removeCommentNodes(root) {
  * @param {Boolean} isRtl True if we are displaying in RTL
  */
 export function convertXmlToBlockly(xmlContainer, isRtl) {
```

Comment:
One other comment -- I think this is ok (removing previous block rendering each time this is called), but it did raise my antenna a little bit (ie, is there a situation where this would now unexpectedly remove previously rendered blocks?) MIght be worth a) updating the description of this function to mention it, and b) checking other uses of this utility to see if its ever used in a way that wouldn't want this new behavior.

Reply:
You're totally right! Apologies for misreading.

Reply:
@mikeharv - Just checked, and this change means I can revert the fixes made in the commit you reference above and the duplicate blocks bug is fixed in `_long_instructions.html.haml` and `_authored_hints.haml`.